test/test_s3.py

```diff
@@ -604,8 +609,7 @@ def test_put_get_with_checksum(self):
         download_checksum_config = S3ChecksumConfig(validate_response=True)
         self._test_s3_put_get_object(download_request, S3RequestType.GET_OBJECT,
                                      checksum_config=download_checksum_config)
-        self.assertTrue(self.did_validate_checksum)
-        self.assertEqual(self.checksum_validation_algorithm, S3ChecksumAlgorithm.CRC32)
+        self.assertTrue(self.done_did_validate_checksum)
         self.assertEqual(HttpHeaders(self.response_headers).get('x-amz-checksum-crc32'),
```

Comment:
why did you take this line out? don't we want to validate that the correct checksum type carried through?
```suggestion
        self.assertTrue(self.done_did_validate_checksum)
        self.assertEqual(self.done_checksum_validation_algorithm, S3ChecksumAlgorithm.CRC32)
        self.assertEqual(HttpHeaders(self.response_headers).get('x-amz-checksum-crc32'),
```

Reply:
Ops, I thought it was checked from the `_validate_successful_response`, turns out we are not invoking it from this test.

Reply:
_validate_successful_response just checks that ANY checksum was set, here we can ensure it was exactly the one we expected